### ISSUE TITLE
geo functions

### ISSUE THREAD

**Issue by mreinsch** [xapix-io/axel-f#42]
add functions to easily work with lat/lang, something like:
 - convert lat/long into geo object
 - convert geo object into lat/long
 - calculate distance between two geo objects
 - add distance to geo objects

**Comment by mreinsch** [comment]
have a look at postgis

**Comment by DG4227** [comment]
cc @oliver-xapix-io

**Comment by oliver-xapix-io** [comment]
@mreinsch @DeLaGuardo @DG4227 Just FYI as inspiration. None of the features have actually been used by customers, no feedback data points available.

In app-rails back in the days we implemented geo points to be an additional data type just like nteger, string, etc. A User could just select that data type from a drop down on the mapping dashboard and on the attribute 2 (!) input fields would appear: One for latitude and one for logitude. The user would then drag & drop the latitude field from Data Source over to latitude input on the Endpoint side. And same for longitude. Only when both were filled the mapping could be saved.

Data was then live transformed using some Ruby geo spatial math library. That would have enabled some basic geo queries like distance and "within radius".

Sidenote that isn't a 100% relevant, as we don't have a "SQL DB Cache" anymore: When a user decided to have transformed data cached in the SQL DB Cache we wrote those geo points to DB with PostGIS extension that stores them in a plain "POINT(lat, long)" format (conversion back and forth to GeoJSON actually gave me some headaches). Plans were to make the PostGIS functions on Geo point data points available as we go, but we stopped working on the MVP codebase then and never got there. User flow for that would have been: User onboards 1-n Data Sources with geo points -> User maps geo points to GeoJSON -> User activates SQL DB Cache feature to cache the data once a day or so -> User can make fancy PostGIS geo queries (like give me all objects within radius) with ultra quick performance.
